jit/src/main/java/org/dcsa/conformance/standards/jit/JitScenarioListBuilder.java

```diff
@@ -1,81 +1,280 @@
 import org.dcsa.conformance.standards.jit.action.SupplyScenarioParametersAction;
-import org.dcsa.conformance.standards.jit.party.JitFilterParameter;
-import org.dcsa.conformance.standards.jit.party.JitRole;
+import org.dcsa.conformance.standards.jit.model.JitTimestampType;
+import org.dcsa.conformance.standards.jit.model.PortCallServiceType;
 
 @Slf4j
 class JitScenarioListBuilder extends ScenarioListBuilder<JitScenarioListBuilder> {
-  private static final ThreadLocal<JitComponentFactory> threadLocalComponentFactory =
-      new ThreadLocal<>();
-  private static final ThreadLocal<String> threadLocalPublisherPartyName = new ThreadLocal<>();
-  private static final ThreadLocal<String> threadLocalSubscriberPartyName = new ThreadLocal<>();
 
   public static LinkedHashMap<String, JitScenarioListBuilder> createModuleScenarioListBuilders(
-      JitComponentFactory componentFactory, String publisherPartyName, String subscriberPartyName) {
-    threadLocalComponentFactory.set(componentFactory);
-    threadLocalPublisherPartyName.set(publisherPartyName);
-    threadLocalSubscriberPartyName.set(subscriberPartyName);
-    return Stream.of(
-            Map.entry(
-                "",
-                noAction()
-                    .thenEither(
-                        scenarioWithParameters(TRANSPORT_CALL_ID, LIMIT),
-                        scenarioWithParameters(VESSEL_IMO_NUMBER, LIMIT),
-                        scenarioWithParameters(CARRIER_SERVICE_CODE, LIMIT),
-                        scenarioWithParameters(UN_LOCATION_CODE, LIMIT),
-                        scenarioWithParameters(OPERATIONS_EVENT_TYPE_CODE, LIMIT),
-                        scenarioWithParameters(
-                            EVENT_CREATED_DATE_TIME_GTE, EVENT_CREATED_DATE_TIME_LT, LIMIT))))
-        .collect(
-            Collectors.toMap(
-                Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e1, LinkedHashMap::new));
+      JitComponentFactory componentFactory, String providerPartyName, String consumerPartyName) {
+    JitScenarioContext context =
+        new JitScenarioContext(providerPartyName, consumerPartyName, componentFactory);
+
+    // Scenario suite: "Service negotiation conformance"
+
+    // Scenario group: "S-ERP-A service types"
+    var scenarioList = new LinkedHashMap<String, JitScenarioListBuilder>();
+    scenarioList.put(
+        "S-ERP-A service types",
+        supplyScenarioParameters(context)
+            .thenEither(
+                PortCallServiceType.getServicesWithERPAndA().stream()
+                    .map(
+                        serviceType ->
+                            portCallService(context, serviceType)
+                                .then(
+                                    sendERPTimestamps(
+                                        context, sendTimestamp(context, JitTimestampType.ACTUAL))))
+                    .toList()
+                    .toArray(new JitScenarioListBuilder[] {})));
```

Comment:
Maybe `.thenEither` could accept a List to avoid this `.toArray` call (which is just extra infrastructure code noise), if we are updating `.core` to be more programmer friendly (I saw the `nonNull` check for creating checks, which is a nice addition).